### ISSUE TITLE
New debug feature for AI scripts

### ISSUE THREAD

**Issue by FS-21** [Phobos-developers/Phobos#53]
AI debugging sometimes is something obscure and terribly slow.

Would be nice (if possible) to have a hotkey to show basic info of a selected unit:
- Teamtype ID.
- ID/step number of the ScriptType that is running and/or the ScriptType ID.
- AITriggerTypes ID (if possible)
- Triggers ID (if possible).
- Unit current mission.

With that information can be more easy to debug/improve AI scripts in multiplayer and single player maps.
The info could be printed in the screen or saved in a log file, depending how loable it is.

**Comment by secsome** [comment]
I tried to make an easy command, however it might be too limited for now.
![202103041450a](https://user-images.githubusercontent.com/41946354/109923364-a44ad280-7cf9-11eb-9c25-413be4b02b8e.png)

**Comment by FS-21** [comment]
Being the first approach is amazing and I started to know what script is being used by the AI.

From what I saw works in 2 ways:

· If executed command and an object is under the mouse cursor it shows that debug info.
· If selected a object but the mouse cursor isn't over anything then shows the debug info of the selected object.

Takes priority the Mouse cursor over the selected object.
I think the timer is too short for watching all the data. Maybe increase 3 times (if possible) would be better.

Noticed too:
· If Human player select multiple (own ) units the last one data is showed.
· All computer players use the same owner.
· Doesn't read the Ares Debug command rulesmd.ini>[GlobalControls]>DebugKeysEnabled=true/false


Show Trigger ID that called that Team ID is possible?

**Comment by FS-21** [comment]
![unknown](https://user-images.githubusercontent.com/1371539/109974463-e411ba00-7cf9-11eb-9d0c-7c86ed6f9054.png)

**Comment by secsome** [comment]
> Welcome secsome!
> Being the first approach is amazing and I started to know what script is being used by the AI.
> 
> From what I saw works in 2 ways:
> 
> · If executed command and an object is under the mouse cursor it shows that debug info.
> · If selected a object but the mouse cursor isn't over anything then shows the debug info of the selected object.
> 
> Takes priority the Mouse cursor over the selected object.
> I think the timer is too short for watching all the data. Maybe increase 3 times (if possible) would be better.
> 
> Noticed too:
> · If Human player select multiple (own ) units the last one data is showed.
> · All computer players use the same owner.
> · Doesn't read the Ares Debug command rulesmd.ini>[GlobalControls]>DebugKeysEnabled=true/false if exists and disables the Debug commands that shouldn't be used by players.
> 
> Show Trigger ID that called that Team ID is possible?
> ![unknown](https://user-images.githubusercontent.com/1371539/109974463-e411ba00-7cf9-11eb-9d0c-7c86ed6f9054.png)

In fact, If more than one object is selected and no TechnoClass is being hovered, this debug only works to show the last member in TechnoClass::Array, and obviously, it's random (for somehow).

For the time the info displayed on the screen, I think it's easy to fix(for now it's 15 game secs), I might fix it after current PR is merged into the develop branch.

This command is just debugging now, so a switch would be added in later implements, this needn't to be worried.
For the last one, I don't think it's quite possible... for now, but a hook to TActionClass might solve it -- howver, too many codes!

**Comment by FS-21** [comment]
I made some changes to the current code in develop branch:
- I left one space between some values and the "(" (Cosmetic change).
- Changed the first owner value by the Country House ID.
- I changed the Script line to look a bit more like the ones in AImd.ini (line number = 1º arg, 2º arg) and in case of line -1 hide those unnecessary variables.
- Multiplied by 4 the message duration, from 150 frames to 600.

![image](https://user-images.githubusercontent.com/1371539/110129278-10940780-7dc8-11eb-862f-479ffbbc0c05.png)


[Small DumpCommand changes 01.zip](https://github.com/Phobos-developers/Phobos/files/6091144/Small.DumpCommand.changes.01.zip)
